snapcraft/parts/plugins/kernel_build.sh

```diff
@@ -0,0 +1,323 @@
+#!/bin/sh
+
+set -eu
+
+# TODO: should these be embedded these in the plugin?
+# Required kernel configs for Ubuntu Core
+_required_generic="
+DEVTMPFS
+DEVTMPFS_MOUNT
+TMPFS_POSIX_ACL
+IPV6
+SYSVIPC
+SYSVIPC_SYSCTL
+VFAT_FS
+NLS_CODEPAGE_437
+NLS_ISO8859_1"
+
+_required_security="
+SECURITY
+SECURITY_APPARMOR
+SYN_COOKIES
+STRICT_DEVMEM
+DEFAULT_SECURITY_APPARMOR
+SECCOMP
+SECCOMP_FILTER"
+
+_required_snappy="
+RD_LZMA
+KEYS
+ENCRYPTED_KEYS
+SQUASHFS
+SQUASHFS_XATTR
+SQUASHFS_XZ
+SQUASHFS_LZO"
+
+_required_systemd="
+DEVTMPFS
+CGROUPS
+INOTIFY_USER
+SIGNALFD
+TIMERFD
+EPOLL
+NET
+SYSFS
+PROC_FS
+FHANDLE
+BLK_DEV_BSG
+NET_NS
+IPV6
+AUTOFS4_FS
+TMPFS_POSIX_ACL
+TMPFS_XATTR
+SECCOMP"
+
+_required_boot="SQUASHFS"
+
+  kver="$(basename "${CRAFT_PART_INSTALL}/modules/"*)"
+
+for arg; do
+  case $arg in
+    defconfig=*)   defconfig=${arg#*=}   ;; # valid: list
+    flavour=*)     flavour=${arg#*=}     ;; # valid: string, trumps kdefconfig
+    configs=*)     configs=${arg#*=}     ;; # valid: list, trumps kconfig-flavour
+    enable-zfs=*)  enable_zfs=${arg#*=}  ;; # valid: list, true|false
```

Comment:
I don't like that this is a kernel option and would rather it be handled as a separate part in the `snapcraft.yaml`... But we can leave it for now.